exercises/input.js

```diff
@@ -50,7 +60,34 @@ class Input{
 		if it is still null, no range has been set, so don't test it
 			if it is not null, then test the range */
 	test(){
-
+		let output = {
+			result: false
+		};
+		let valueTest = this.inputElement.val();
+		
+		if (this.pattern.test(valueTest)){
+			let rangeTest = this.getRange();
+			if(!rangeTest.min && !rangeTest.max){
+				output.result = true;
+				return output;
+			}
+			if (rangeTest.min){
```

Comment:
I kinda understand this, but you might as well combine the existence test with the range test

like

```
if( this.getRange().min !== null && valueTest < this.getRange().min )
```

Reply:
It would have been fewer tests with the same results and less repeated code, I think